src/python/pants/option/global_options.py

```diff
@@ -549,11 +549,12 @@ class BootstrapOptions:
         help="Re-resolve plugins, even if previously resolved.",
     )
     level = EnumOption(
-        "-l",
         "--level",
         default=LogLevel.INFO,
         daemon=True,
         help="Set the logging level.",
+        # This is the only option where we allow a short flag name.
+        _unsafe_short_flag_name="-l",
```

Comment:
We could subclass `EnumOption` for `LevelOption` and then set the `flag_names` in `__new__`.
Then remove `_unsafe_short_flag_name` althogether.

 Just a thought.

Reply:
Great idea :)

Reply:
I'd also be fine deprecating `-l` and requiring `--level=DEBUG` or whatever, it doesn't seem like a big burden.

Reply:
I like `-ldebug` shorthand personally. 

But also that change would involve deprecation, and this change can go in now, so this PR is good-to-go regardless of that decision, right?

Reply:
Yeah, I've approved it, I just think `-ldebug` is not worth jumping through hoops for, and I'd be fine deprecating and removing it.